client/analytics/report/coupons/table.js

```diff
@@ -67,22 +68,29 @@ export default class CouponsReportTable extends Component {
 	}
 
 	getRowsContent( coupons ) {
+		const { query } = this.props;
+		const persistedQuery = getPersistedQuery( query );
+
 		return map( coupons, coupon => {
 			const { amount, coupon_id, extended_info, orders_count } = coupon;
 			const { code, date_created, date_expires, discount_type } = extended_info;
 
-			// @TODO must link to the coupon detail report
+			const productUrl = getNewPath( persistedQuery, '/analytics/coupons', {
```

Comment:
Nice! Minor but should this const be called `couponsUrl` perhaps?

Reply:
Of course! :man_facepalming: